module-2/lab-probability-distribution/your_code/main2.ipynb

```diff
@@ -0,0 +1,561 @@
+{
```

Comment:
- Now that you have finished the exercise, try to do the following in a couple of lines: 
```Python
mean=50
sigma=5
sample_size=[10, 50, 500, 5000]


s10= np.random.normal(mean, sigma,10)
s50= np.random.normal(mean, sigma,50)
s100= np.random.normal(mean, sigma,100)
s5000= np.random.normal(mean, sigma,5000)

print(stats.normaltest(s1))
print(stats.normaltest(s50))
print(stats.normaltest(s100))
print(stats.normaltest(s5000))
```
- Sometimes it's useful to seed the random number generator. Seeding a pseudo-random number generator gives it its first "previous" value. Each seed value will correspond to a sequence of generated values for a given random number generator. That is, if you provide the same seed twice, you get the same sequence of numbers twice. https://machinelearningmastery.com/how-to-generate-random-numbers-in-python/
- The following isn't quite correct: "Interpret the normaltest results and make your observations.
No se demuestra que a mayor número de muestras el p value sea mayor, de hecho el mayor valor se da en la muestra de 10 elementos". You can only check if the statistic is above or below the significance level you have pre-defined (your alpha parameter for a certain confidence level). It's not a good idea to check the level of the statistic. 
-Normality tests are quite sensitive to sample size thus for a sample size below a couple of thousands it should be a better idea to evaluate normality through a plot, in particular a quantile-quantile plot (QQ plot). A Q-Q plot is a scatterplot created by plotting two sets of quantiles against one another. It plots the quantiles of our sample vs the theoretical quantiles of the distribution type we want to test (this case a normal distribution) How to do this in Python? https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.probplot.html.
-There are 2 other normality tests (additional to D'Agostino k^2 test, normaltest in Python) that are very used in to test normality of our data. These are Kolmogorov-Smirnof and Shapiro-Wilk tests. I encourage you to deepen on these type of tests. In case you are interested, I can provide you some useful resources.
- In Spanish: Cuando corremos un test de contraste de hipótesis, si los estadísticos de contraste están dentro del rango que le hemos establecido decimos que "no podemos rechazar la hipótesis nula". Esto lo decimos porque a pesar de un resultado positivo del estadístico de contraste podría no ser normal. Esto lo ves claramente en el primer histograma tamaño de la muestra 5, el p-value te dice que no puedes rechazar la hipótesis nula de normalidad y, sin embargo, el histograma muestra algo que no tiene nada que ver con una normal. Aquí podríamos graficar un gráfico de quantiles y, probablemente, veríamos, al igual que en el histograma, que los datos no vienen de una distribución normal.
- Good one!! "Cuanto mayor es el tamaño de la muestra, más se aproxima a una distribución normal".  Interpretation of the CLT: take a look on these 2 articles (the most clear explanations I've found). You can replicate in Python some of the examples they include.
https://towardsdatascience.com/understanding-the-central-limit-theorem-642473c63ad8.
https://www.statisticshowto.datasciencecentral.com/probability-and-statistics/normal-distributions/central-limit-theorem-definition-examples/
- Regarding this "What did you notice from the distribution plots? Comment your findings.
A medida que aumenta la muestra la probabilidad de toda la muestra haga click en un anuncio es cero. Por otro lado la probabilidad de los que hacen click es cada vez menor", it's not quite correct. The important conclusion here is that as larger the sample size more similar to a normal distribution. Why? 
Normal distribution can be used as an approximation of Binomial and Poisson distributions under certain circumstances. Generally speaking:
Binomial: X ~ B(n, p) and if n is large and/or p is close to ½, then X is approximately N(np, npq)
Poisson: X ~ Po(l) then for large values of l, X ~ N(l, l) approximately.
However, we must be careful because binomial and poisson distributions are discrete while Normal distribution is continuous. We must use a continuity correction.
- Take a look on the plot for poisson, it looks like a normal distribution, don't you think so?